packages/react-components/src/components/MicrophoneButton.tsx

```diff
@@ -56,6 +57,24 @@ export const MicrophoneButton = (props: MicrophoneButtonProps): JSX.Element => {
   const localeStrings = useLocale().strings.microphoneButton;
   const strings = { ...localeStrings, ...props.strings };
 
+  const theme = useTheme();
+  const onRenderMicOnIcon = (): JSX.Element => (
+    <Icon
```

Comment:
(1) Line 46/47 are now dead code?

(2) Defining these inline will cause extra renders for the icons. Consider using `useMemo` to only render when `them` or `props.disabled` changes.

(3) Reading the code below, it is not obvious to me why setting the theme colors fixes the high contrast issue. Could you:
  (a) Add a comment in the code explaining why the theme colors are needed (perhaps also what the default behavior was before adding these)
  (b) Extract a helper component since this is needed for all buttons (this also takes care of the extra rendering issue):

```ts

const HighContrastAwareIcon(props: { iconName: string, disabled: boolean}): JSX.Element => {
    const theme = useTheme();
    // Comment here describing why we need to set the colors from theme.
    return <Icon iconName={props.iconName} className={.... etc. }>
}
```